Controllers/Backend/MollieOrders.php

```diff
@@ -334,14 +335,7 @@ public function refundAction()
                 ]
             );
 
-            # switch to the config and api key of the shop from the order
-            $shopSwitcher = new MollieShopSwitcher($this->container);
-            $this->config = $shopSwitcher->getConfig($order->getShop()->getId());
-            $this->apiClient = $shopSwitcher->getMollieApi($order->getShop()->getId());
-
-            $transaction = $this->orderService->getOrderTransactionByNumber($order->getNumber());
-
-            $refund = $this->refundService->refundFullOrder($order, $transaction);
+            $refund = $this->refundOrderFacade->refundOrder($order->getNumber(), $customAmount);
```

Comment:
attention, you are removing the config and api switching!
in the backend, orders can come from any subshop...so this is always related to the provided order that will be refunded. i think this breaks things